### PR TITLE
Ignore reply packets from other PDs

### PR DESCRIPTION
**Pull request by prxvm**
I was attempting to use two PDs with this library, when I ran across some errors. The packet coming from a different PD is marked as an error condition and the secure session gets discarded. These changes attempt to overcome these errors.

### CODE REVIEW

File: src/osdp_pd.c [hunk 1]
```diff
@@ -983,7 +987,18 @@ static void osdp_pd_update(struct osdp_pd *pd)
 	case OSDP_PD_STATE_PROCESS_CMD:
 		ret = pd_receve_packet(pd);
 		if (ret == OSDP_PD_ERR_NO_DATA &&
-		    osdp_millis_since(pd->tstamp) < OSDP_RESP_TOUT_MS) {
+			osdp_millis_since(pd->tstamp) < OSDP_RESP_TOUT_MS) {
+			break;
+		}
+		if (ret == OSDP_PD_ERR_IGNORE) {
+			/* Ignore this message */
+			pd->state = OSDP_PD_ERR_NONE;
```
Comment (sidcha) [inline src/osdp_pd.c:995]: These two lines are also not needed; `pd->state` is overwritten in the next IF-ESLE block anyway.

File: src/osdp_pd.c [hunk 2]
```diff
@@ -954,8 +958,8 @@ static int pd_receve_packet(struct osdp_pd *pd)
 	remaining = pd->rx_buf_len - len;
 	if (remaining) {
 		memmove(pd->rx_buf, pd->rx_buf + len, remaining);
-		pd->rx_buf_len = remaining;
 	}
+	pd->rx_buf_len = remaining;
```
Comment (sidcha) [inline src/osdp_pd.c:962]: This is a separate bug fix :)

nice catch!

**Review by sidcha — CHANGES_REQUESTED** [review]
@prxvm thanks for the fixes, very helpful. :) 

The commit message needs to be changed. The git ideology is to have the first line of the commit message as a short subject line (usually within 60 to 80 chars describing your change like this PRs title) and then one blank line and then your elaborate commit message. For example, for this PR, this would be a fantastic commit message:
```
Ignore packet that comes as a reply from another PD.

If there is more data the 'skipped' packet than the entire packet itself,
store in rx_buf_len and adjust state based on it
```

PS. If you are new to git, you can make the changes suggested in the review, `git add -A` and then use `git commit --amend` change the commit message and `git push -f origin <your-branch>` to update this PR.

File: src/osdp_pd.c (hunk 2 above)
Comment (prxvm) [reply src/osdp_pd.c]: If you don't mind can we keep this as part of this commit. It is required for this patch to work as expected.

**Comment by prxvm** [comment]
Thank you @sidcha for your excellent, kind review. I have addressed your comments.

File: src/osdp_pd.c (hunk 2 above)
Comment (sidcha) [reply src/osdp_pd.c]: I just wanted to highlight that fix. Of course it can stay here.

**Review by sidcha — APPROVED** [review]
Looks perfect, thanks for the PR!

**Pull request merged**